sync/sync.go

```diff
@@ -54,36 +54,23 @@ func (s *Synchronizer) fetcherTask(ctx context.Context, height uint64, verifiers
 				continue
 			}
 
-			// There are classes in deployed transactions which refer to class hash that are no present in declared
-			// classes. Thus, we need to fetch all the classes which are referenced in deployed contracts
-			referencedClasses := make(map[felt.Felt]core.Class)
-			for _, deployedContract := range stateUpdate.StateDiff.DeployedContracts {
-				referencedClasses[*deployedContract.ClassHash] = nil
-			}
-			for _, classHash := range stateUpdate.StateDiff.DeclaredV0Classes { // todo: fetch v1 classes as well
-				referencedClasses[*classHash] = nil
-			}
-			for classHash := range referencedClasses {
-				class, err := s.StarknetData.Class(ctx, &classHash)
-				if err != nil {
-					continue
-				}
-				referencedClasses[classHash] = class
-			}
-
```

Comment:
By moving this to the blockchain we are not conflating what `fetcherTask` and `verifierTask` are supposed to do.

Reply:
I know it is not ideal, feel free to suggest an alternative

Reply:
This made me realize that we have cases where the verification process halts without no real reason.

We start fetchers and in their callback functions they start the verifiers. Since fetcher callbacks are executed in order, verification tasks are started in order as well. This means that without block N being downloaded, we cannot start verifying block N+1. Which is no ideal, verification should be able to start right away for each individual block after it is downloaded.

Reply:
On the other hand, this behavior might make sense because this avoids using up the verifiers tasks for later blocks while the block `HEAD+1` is waiting for a verifier to be available

Reply:
A suggestion would be, instead of fetching unknown classes, we can fetch unknown class hashes, then fetch the classes in the sync package. Might just be a tad bit slower.

Reply:
Now the question is how does `blockchain` know that `sync` successfully fetched all the missing classes.